dotCMS/src/main/java/com/dotcms/rest/IntegrityDataRequestChecker.java

```diff
@@ -0,0 +1,173 @@
+import com.dotmarketing.exception.DotHibernateException;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.language.LanguageException;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.portal.model.User;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+/**
+ * Thread to constantly check the integrity check status by hitting on receiver host endpoint.
+ * Anything but {@link com.dotcms.rest.IntegrityResource.ProcessStatus}.PROCESSING is considered that the integrity data
+ * generation process is done.
+ */
+public class IntegrityDataRequestChecker implements Runnable {
+
+    private final String authToken;
+    private final PublishingEndPoint endpoint;
+    private final String integrityDataRequestId;
+    private final HttpSession session;
+    private final InitDataObject initData;
+
+    public IntegrityDataRequestChecker(String authToken,
+                                       PublishingEndPoint endpoint,
+                                       String integrityDataRequestId,
+                                       HttpSession session,
+                                       InitDataObject initData) {
+        this.authToken = authToken;
+        this.endpoint = endpoint;
+        this.integrityDataRequestId = integrityDataRequestId;
+        this.session = session;
+        this.initData = initData;
+    }
+
+    @Override
+    public void run() {
+        final Map<String, String> paramsMap = initData.getParamsMap();
+        final String endpointId = paramsMap.get("endpoint");
+        final User loggedUser = initData.getUser();
+        final FormDataMultiPart form = new FormDataMultiPart();
+        form.field("AUTH_TOKEN",authToken);
+        form.field("REQUEST_ID", integrityDataRequestId);
+        final String url = endpoint.toURL() + "/api/integrity/getintegritydata/";
+
+        boolean processing = true;
+        while(processing) {
+            final Response response = IntegrityResource.postWithEndpointState(
+                    endpoint.getId(),
+                    url,
+                    new MediaType("application", "zip"),
+                    Entity.entity(form, form.getMediaType()));
+
+            if (response.getStatus() == HttpStatus.SC_OK) {
+                processing = false;
+
+                final String outputDir = IntegrityUtil.getIntegrityDataPath(endpoint.getId());
+                final InputStream zipFile = response.readEntity(InputStream.class);
+
+                try {
+                    IntegrityUtil.unzipFile(zipFile, outputDir);
+                } catch(Exception e) {
+                    //Special handling if the thread was interrupted
+                    if (e instanceof InterruptedException) {
```

Comment:
Codacy found an issue: [An instanceof check is being performed on the caught exception.  Create a separate catch clause for this exception type.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6151243)